src/cbc_syslog/util/output.py

```diff
@@ -0,0 +1,118 @@
+
+# *******************************************************
+# Copyright (c) VMware, Inc. 2020-2023. All Rights Reserved.
+# SPDX-License-Identifier: MIT
+# *******************************************************
+# *
+# * DISCLAIMER. THIS PROGRAM IS PROVIDED TO YOU "AS IS" WITHOUT
+# * WARRANTIES OR CONDITIONS OF ANY KIND, WHETHER ORAL OR WRITTEN,
+# * EXPRESS OR IMPLIED. THE AUTHOR SPECIFICALLY DISCLAIMS ANY IMPLIED
+# * WARRANTIES OR CONDITIONS OF MERCHANTABILITY, SATISFACTORY QUALITY,
+# * NON-INFRINGEMENT AND FITNESS FOR A PARTICULAR PURPOSE.
+
+"""Output class"""
+
+import logging
+import requests
+import socket
+import ssl
+import traceback
+
+log = logging.getLogger(__name__)
+
+
+class Output:
+    """Output mechanisms to send data to local or external destination"""
+
+    def __init__(self, output_params):
+        """
+        Initialize the Output object.
+
+        Args:
+            output_params (dict): The output configuration
+
+        Example:
+            {
+                "type": str,
+                "host": str,
+                "port": str,
+                "tls_verify": bool,
+                "http_headers": dict,
+                "ca_cert": str,
+                "cert": str,
+                "key": str,
+                "key_password": str
+            }
+        """
+        self.output_params = output_params
```

Comment:
Should we make a copy of this or it is safe to directly assign it?

Reply:
It's not intended to be modified in this class if a property needs manipulated then yeah a copy would be made